src/linear_algebra/matrix/hermitian.lean

```diff
@@ -145,6 +145,14 @@ section is_R_or_C
 
 variables [is_R_or_C α] [is_R_or_C β]
 
+/-- The diagonal elements of a complex hermitian matrix are real. -/
+lemma complex_hermitian_real_diagonal {A : matrix n n α} (h : A.is_hermitian) :
+  ∀ i, ∃ (r : ℝ), (A i i) = r :=
+begin
+  intro i, rw ←is_R_or_C.eq_conj_iff_real, unfold is_hermitian at h, nth_rewrite 0 ←h,
+  rw [matrix.conj_transpose_apply, is_R_or_C.star_def, star_ring_end_self_apply],
+end
```

Comment:
```suggestion
lemma is_hermitian.coe_re_apply_self {A : matrix n n α} (h : A.is_hermitian) (i : n) :
  (re (A i i) : α) = A i i :=
by rw [←eq_conj_iff_re, ←star_def, ←conj_transpose_apply, h.eq]

/-- The diagonal elements of a complex hermitian matrix are real. -/
lemma is_hermitian.coe_re_diag {A : matrix n n α} (h : A.is_hermitian) :
  (λ i, (re (A.diag i) : α)) = A.diag :=
funext h.coe_re_apply_self
```
A few notes about my suggested changes:
1. By naming within the `matrix.is_hermitian` namespace, we enable dot notation. See an example of dot notation at work with `h.eq` (which is really `matrix.is_hermitian.eq h`).
2. The rest of the name here follows our [naming conventions](https://leanprover-community.github.io/contribute/naming.html). I would have called your original statement `is_hermitian.exists_real_apply_self_eq` or something like that.
3. Generally we move all hypothesis to the left of the colon, as per the [code style guidelines](https://leanprover-community.github.io/contribute/style.html#hypotheses-left-of-colon).
4. I used `is_R_or_C.re` so that you don't have to hide it behind the existential quantifier, it's better to be explicit when possible.
5. The left-hand side of an equality should generally be the "less simple" side, so I've put it on the left.
6. I added a separate lemma for convenience to link this to `matrix.diag`.

Finally, if you're using `unfold` it generally means that you are missing an existing API lemma (or it should exist but doesn't, but having missed it is often more likely). In this case, that lemma was `matrix.is_hermitian.eq`.